sdk/metricsadvisor/azure-ai-metricsadvisor/azure/ai/metricsadvisor/_patch.py

```diff
@@ -1,161 +1,665 @@
+        """
+        subscription_key = kwargs.pop("subscription_key", None)
+        api_key = kwargs.pop("api_key", None)
+        if len(kwargs) > 0:
+            raise TypeError("Got an unexpected keyword argument: {}".format(list(kwargs.keys())[0]))
+        if subscription_key:
+            if not isinstance(subscription_key, str):
+                raise TypeError("The subscription_key used for updating must be a string.")
+            self._subscription_key = subscription_key
+        if api_key:
+            if not isinstance(api_key, str):
+                raise TypeError("The api_key used for updating must be a string.")
+            self._api_key = api_key
+
+
+class MetricsAdvisorKeyCredentialPolicy(SansIOHTTPPolicy):
+    """Adds a key header for the provided credential.
+
+    :param credential: The credential used to authenticate requests.
+    :type credential: ~azure.core.credentials.AzureKeyCredential
+    :param str name: The name of the key header used for the credential.
+    :raises: ValueError or TypeError
+    """
+
+    def __init__(
+        self, credential: MetricsAdvisorKeyCredential, **kwargs: Any  # pylint: disable=unused-argument
+    ) -> None:
+        super(MetricsAdvisorKeyCredentialPolicy, self).__init__()
+        self._credential = credential
+
+    def on_request(self, request):
+        request.http_request.headers[_API_KEY_HEADER_NAME] = self._credential.subscription_key
+        request.http_request.headers[_X_API_KEY_HEADER_NAME] = self._credential.api_key
+
+
+def get_authentication_policy(credential):
+    authentication_policy = None
+    if credential is None:
+        raise ValueError("Parameter 'credential' must not be None.")
+    if isinstance(credential, MetricsAdvisorKeyCredential):
+        return MetricsAdvisorKeyCredentialPolicy(credential)
+    if credential is not None and not hasattr(credential, "get_token"):
+        raise TypeError(
+            "Unsupported credential: {}. Use an instance of MetricsAdvisorKeyCredential "
+            "or a token credential from azure.identity".format(type(credential))
+        )
+
+    return authentication_policy
+
+
+class MetricsAdvisorClient:  # pylint: disable=client-accepts-api-version-keyword
+    """Represents an client that calls restful API of Azure Metrics Advisor service.
+    :param str endpoint: Supported Cognitive Services endpoints (protocol and hostname,
+        for example: https://:code:`<resource-name>`.cognitiveservices.azure.com).
+    :param credential: An instance of ~azure.ai.metricsadvisor.MetricsAdvisorKeyCredential.
+        which requires both subscription key and API key. Or an object which can provide an access
+        token for the vault, such as a credential from :mod:`azure.identity`
+    :type credential: ~azure.ai.metricsadvisor.MetricsAdvisorKeyCredential or ~azure.core.credentials.TokenCredential
+    """
+
+    def __init__(self, endpoint: str, credential: MetricsAdvisorKeyCredential, **kwargs: Any) -> None:
```

Comment:
I think credential should be a Union[MetricsAdvisorKeyCredential, TokenCredential]

Reply:
Yeah i guess it does accept both, I just followed current typing. Changing now